netspeed/src/netspeed.c

```diff
@@ -1233,11 +1233,11 @@ showinfo_cb(GtkAction *action, gpointer data)
 	gtk_grid_attach(GTK_GRID(grid), applet->outbytes_text, 3, 2, 1, 1);
 
 	/* check if we got an ipv6 address */
-	if (applet->devinfo.ipv6 && (strlen (applet->devinfo.ipv6) > 2)) {
+	if (strlen (applet->devinfo->ipv6) > 2) {
```

Comment:
it isn't needed condition `if (applet->deviceinfo->ipv6)` to check if it is `NULL` ?

Reply:
No, it isn't.

Reply:
can you explain why?

`strlen` with `NULL` makes undefined behavior, where is the check if it is `NULL` ?

Reply:
Take a look at the variable declaration, for this reason it cannot be null.

Reply:
```
$ grep -n ipv6 netspeed/src/backend.h
66:	char ipv6[INET6_ADDRSTRLEN];
```